test/integration/multipleDepositWithdraw.js

```diff
@@ -0,0 +1,326 @@
+    ]);
+    // initialize anchors on both chains
+    OriginChainLinkableAnchorInstance = await LinkableAnchorContract.new(
+      verifier.address,
+      hasher.address,
+      tokenDenomination,
+      merkleTreeHeight,
+      originChainID,
+      originChainToken.address,
+    {from: sender});
+    DestChainLinkableAnchorInstance = await LinkableAnchorContract.new(
+      verifier.address,
+      hasher.address,
+      tokenDenomination,
+      merkleTreeHeight,
+      destChainID,
+      destChainToken.address,
+    {from: sender});
+    // create resource ID using anchor address
+    resourceID = Helpers.createResourceID(OriginChainLinkableAnchorInstance.address, 0);
+    initialResourceIDs = [resourceID];
+    destInitialContractAddresses = [OriginChainLinkableAnchorInstance.address];
+    // initialize anchorHanders 
+    await Promise.all([
+      AnchorHandlerContract.new(DestBridgeInstance.address, initialResourceIDs, destInitialContractAddresses)
+        .then(instance => DestAnchorHandlerInstance = instance),
+    ]);
+    // increase allowance and set resources for bridge
+    await Promise.all([
+      originChainToken.approve(OriginChainLinkableAnchorInstance.address, initialTokenMintAmount, { from: sender }),
+      destChainToken.approve(DestChainLinkableAnchorInstance.address, initialTokenMintAmount, { from: sender }),
+      DestBridgeInstance.adminSetResource(DestAnchorHandlerInstance.address, resourceID, DestChainLinkableAnchorInstance.address)
+    ]);
+     // set bridge and handler permissions for anchor
+     await Promise.all([
+      DestChainLinkableAnchorInstance.setHandler(DestAnchorHandlerInstance.address, {from: sender}),
+      DestChainLinkableAnchorInstance.setBridge(DestBridgeInstance.address, {from: sender})
+     ]);
+
+    createWitness = async (data) => {
+      const wtns = {type: "mem"};
+      await snarkjs.wtns.calculate(data, path.join(
+        "artifacts/circuits",
+        "bridge",
+        "poseidon_bridge_2.wasm"
+      ), wtns);
+      return wtns;
+    }
+
+    tree = new MerkleTree(merkleTreeHeight, null, prefix)
+    zkey_final = fs.readFileSync('build/bridge2/circuit_final.zkey').buffer;
+  });
+
+  it('[sanity] dest chain bridge configured with threshold and relayers', async () => {
+    assert.equal(await DestBridgeInstance._chainID(), destChainID)
+    assert.equal(await DestBridgeInstance._relayerThreshold(), relayerThreshold)
+    assert.equal((await DestBridgeInstance._totalRelayers()).toString(), '2')
+  })
+
+  it.only('withdrawing across bridge after two deposits should work', async () => {
+    // minting Tokens
+    await originChainToken.mint(sender, initialTokenMintAmount);
+    // deposit on both chains and define nonces based on events emmited
+    let { logs } = await OriginChainLinkableAnchorInstance.deposit(
+      Helpers.toFixedHex(Helpers.generateDeposit(destChainID).commitment), {from: sender});
+    originUpdateNonce = logs[0].args.leafIndex;
```

Comment:
Update nonce isn't necessarily this. A nonce is a value we should take to only increment by 1 upon each action.

Since it is possible that anchors are not updated after every deposit, it is better to take the update nonce to be 1 + previous update nonce of that anchor/contract.

It suffices for now but I do want to make sure we nail this down now.